CMakeLists.txt

```diff
@@ -318,6 +318,8 @@ if(WIN32)
   else()
     try_append_cxx_flags("-Wa,-mbig-obj" TARGET core_interface_debug SKIP_LINK)
   endif()
+
+  target_link_libraries(core_interface INTERFACE bcrypt)
```

Comment:
We might need to update https://github.com/bitcoin/bitcoin/blob/master/contrib/devtools/symbol-check.py#L156
(likely add a DLL, possibly even remove one, though i doubt this was all we use from ADVAPI32).
The guix build will tell.

Reply:
as expected:
```
unning symbol and dynamic library checks...
bcrypt.dll is not in ALLOWED_LIBRARIES!
```

Reply:
Why not narrow the scope of this dependency to the `bitcoin_util` and `bitcoinkernel` targets?

Reply:
Fixed both issues, thanks.

Actually, it turns out `symbol-check.py` actually complains on master about bcrypt! That's because it gets included by secp, I've opened https://github.com/bitcoin/bitcoin/pull/32431 which includes the same change and also bumps the lief version, I'll rebase (if) whichever of these two PR's get merged first.